api/registry.py

```diff
@@ -64,6 +68,24 @@ def for_protocol_and_goal(cls, _db, protocol, goal):
         ):
             yield cls(i)
 
+    @classmethod
+    def for_protocol_goal_and_url(cls, _db, protocol, goal, url):
+        """Get a LibraryRegistry for the given protocol, goal, and
+        URL. Create the corresponding ExternalIntegration if necessary.
+        """
+        try:
+            integration = ExternalIntegration.with_setting_value(
+                _db, protocol, goal, ExternalIntegration.URL, url
+            ).one()
+        except NoResultFound:
+            integration = None
```

Comment:
should you also handle multiple results found?

Reply:
one() will raise an exception if there are multiple results, which I think is appropriate.